plugins/default-browser-emulator/injected-scripts/console.ts

```diff
@@ -1,48 +1,211 @@
-export type Args = {
-  mode: 'disableConsole' | 'patchLeaks';
-};
+export type Args = never;
 
-const typedArgs = args as Args;
+// Ported from chromium/src/v8/src/builtins/builtins-console.cc
+type ConsoleKeys = Array<keyof typeof console>;
+const CONSOLE_METHODS_WITH_FORMATTING: ConsoleKeys = [
+  'debug',
+  'error',
+  'info',
+  'log',
+  'warn',
+  'trace',
+  'group',
+  'groupCollapsed',
+  'assert',
+] as const;
 
-ObjectCached.keys(console).forEach(key => {
-  proxyFunction(console, key, (target, thisArg, args) => {
-    if (typedArgs.mode === 'disableConsole') return undefined;
-    args = replaceErrorStackWithOriginal(args);
-    return ReflectCached.apply(target, thisArg, args);
+const CONSOLE_METHODS: ConsoleKeys = [
+  'dir',
+  'dirxml',
+  'table',
+  'groupEnd',
+  'clear',
+  'count',
+  'countReset',
+  'profile',
+  'profileEnd',
+  'timeLog',
+] as const;
+
+const ALL_CONSOLE_METHODS: ConsoleKeys = [
+  ...CONSOLE_METHODS_WITH_FORMATTING,
+  ...CONSOLE_METHODS,
+] as const;
+
+ALL_CONSOLE_METHODS.forEach(key => {
+  if (typeof console[key] !== 'function') return;
+  replaceFunction(console, key, (target, thisArg, argArray) => {
+    if (CONSOLE_METHODS_WITH_FORMATTING.includes(key)) {
+      formatter(argArray);
+    }
+
+    // Don't use map here, or it will leak in stacktraces
+    for (let idx = 0; idx < argArray.length; idx++) {
+      argArray[idx] = V8ValueStringBuilder.toString(argArray[idx]);
+    }
+    return ReflectCached.apply(target, thisArg, argArray);
   });
 });
 
-const defaultErrorStackGetter = Object.getOwnPropertyDescriptor(new Error(''), 'stack').get;
+// Ported from chromium/src/v8/src/builtins/builtins-console.cc
+function formatter(args: any[]) {
```

Comment:
Do we really need to include this entire thing? Do we care that the deep formatting happens? Seems like we only really care about the accessing

Reply:
All of this has very subtle side effects and skipping stuff seemed to trigger anti bots tests, so will leave the entire thing here just to be safe